src/Illuminate/View/Compilers/BladeCompiler.php

```diff
@@ -579,7 +579,11 @@ protected function compileFor($expression)
      */
     protected function compileForeach($expression)
     {
-        return "<?php foreach{$expression}: ?>";
+        preg_match('/\( *(.*) *as /', $expression, $matches);
+
+        $content = trim($matches[1]);
+
+        return "<?php \$__env->addLoop(count({$content})); foreach{$expression}: \$__env->incrementLoopIndices(); ?>";
```

Comment:
You should only `count` the object if it is `Countable`. In the case that it's not countable, revIndex/isLast/size will not be available, but the remaining functionality (isFirst, index) can still be used.